server/src/main/java/org/opensearch/index/engine/exec/format/DataFormat.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.engine.exec.format;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.mapper.ParametrizedFieldMapper;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a data format.
+ */
+@ExperimentalApi
+public interface DataFormat {
+
+    /**
+     *
+     * @return name identifier for the data format.
+     */
+    String name();
+
+    /**
+     * Index level settings supported by this data format.
+     */
+    default Settings dataFormatSettings() {
```

Comment:
Why we need two different functions for fetching the Settings? Currently all the settings comes from 
```
    public List<Setting<?>> getPluginSettings() {
        return plugins.stream().flatMap(p -> p.v2().getSettings().stream()).collect(Collectors.toList());
    }
```
 and later on filtered [here](https://github.com/opensearch-project/OpenSearch/blob/aca2e9de7daaadee77c33e21ff5215c3f1e8600f/server/src/main/java/org/opensearch/common/settings/SettingsModule.java#L234)